File: performance/option_operations.lua
math.randomseed(42)
local success_count = 0
local none_count = 0

for i = 1, 50000 do
  local val = math.random(0, 100)
  local opt = val > 50 and val or nil

  if opt then
    success_count = success_count + opt
  else
    none_count = none_count + 1
  end
end

print(success_count)
print(none_count)

File: performance/option_operations.py
import random

def main():
    random.seed(42)
    success_count = 0
    none_count = 0

    # Option operations benchmark (simulated with None)
    for _ in range(50000):
        val = random.randint(0, 100)

        # Create option based on value
        opt = val if val > 50 else None

        # Pattern match and process
        if opt is not None:
            success_count = success_count + opt
        else:
            none_count = none_count + 1

    print(success_count)
    print(none_count)

main()
